accessors/datastream/datastream_accessor.go

```diff
@@ -0,0 +1,124 @@
+	DeleteConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, id string, projectId string, region string) error
+	GetConnProfilesRegion(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectId string, region string) ([]string, error)
+	CreateConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, req *datastreampb.CreateConnectionProfileRequest) (*datastreampb.ConnectionProfile, error)
+	ConnectionProfileExists(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectId string, profileName string, profileLocation string, connectionProfiles map[string][]string) (bool, error)
+}
+type DatastreamAccessorImpl struct {}
+
+
+// FetchTargetBucketAndPath fetches the bucket and path name from a Datastream destination config.
+func (da *DatastreamAccessorImpl) FetchTargetBucketAndPath(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, projectID string, datastreamDestinationConnCfg streaming.DstConnCfg) (string, string, error) {
+	if datastreamClient == nil {
+		return "", "", fmt.Errorf("datastream client could not be created")
+	}
+	dstProf := fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamDestinationConnCfg.Location, datastreamDestinationConnCfg.Name)
+	// `GetConnectionProfile` has out of box retries. Ref - https://github.com/googleapis/googleapis/blob/master/google/cloud/datastream/v1/datastream_grpc_service_config.json
+	res, err := datastreamClient.GetConnectionProfile(ctx, dstProf)
+	if err != nil {
+		return "", "", fmt.Errorf("could not get connection profiles: %v", err)
+	}
+	// Fetch the GCS path from the target connection profile.
+	// The Get calls for Google Cloud Storage API have out of box retries.
+	// Reference - https://cloud.google.com/storage/docs/retry-strategy#idempotency-operations
+	gcsProfile := res.Profile.(*datastreampb.ConnectionProfile_GcsProfile).GcsProfile
+	bucketName := gcsProfile.Bucket
+	prefix := gcsProfile.RootPath + datastreamDestinationConnCfg.Prefix
+	prefix = utils.ConcatDirectoryPath(prefix, "data/")
+	return bucketName, prefix, nil
+}
+
+// Deletes a connection Profile
+func (da *DatastreamAccessorImpl) DeleteConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, id string, projectId string, region string) error {
+	op, err :=datastreamClient.DeleteConnectionProfile(ctx,  &datastreampb.DeleteConnectionProfileRequest{
+		Name: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectId, region, id),
+	})
+	if err != nil {
+		return err
+	}
+	
+	err = op.Wait(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates new connection Profile
+func (da *DatastreamAccessorImpl) CreateConnectionProfile(ctx context.Context, datastreamClient datastreamclient.DatastreamClient, req *datastreampb.CreateConnectionProfileRequest) (*datastreampb.ConnectionProfile, error) {
```

Comment:
What we had decided for accessors was that we should avoid request bodies as function parameters. Instead, we create this request inside the function while the parameters just take the required strings/data etc. Like what is done in DataflowAccessorImpl.LaunchDataflowTemplate().

Reply:
In this particular case it becomes a bit complicated as is it possible to have multiple types of connection profiles(eg gcs, mysql, etc) and all of them require substantially different parameters. In either case I would need to create a struct to be passed as types may differ which raises the question for this particular case why not pass the request only?

Reply:
Fair makes sense in this case, its not a strict rule but we should do our best to adhere to that.